src/fs.ts

```diff
@@ -78,6 +78,33 @@ export class LocalFileSystem implements FileSystem {
 	}
 }
 
+export class MapFileSystem implements FileSystem {
+
+	constructor(private files: Map<string, string>) {}
+
+	/**
+	 * Converts the URI to an absolute path
+	 */
+	protected resolveUriToPath(uri: string): string {
+		return uri2path(uri).substring(1);
```

Comment:
Two different URIs like `file:///1` and `inmemory:/1` would override each other and return the same content